src/system.ts

```diff
@@ -411,6 +415,12 @@ export const chunkDataSource =
         chunkDataStore: new FsChunkDataStore({ log, baseDir: 'data/chunks' }),
       });
 
+export const chunkDataSource =
+  config.CHUNK_DATA_SOURCE_TYPE === 'legacy-s3' &&
```

Comment:
Coupling this to the legacy implementations is unnecessarily restrictive. The `CompositeChunkSource` should be instantiated as long as `chunkMetaDataSource` and `chunkDataSource` are defined.